consent/api/v1/views.py

```diff
@@ -129,19 +146,43 @@ def get_required_query_params(self, request):
                 self.REQUIRED_PARAM_COURSE_ID,
                 request.query_params.get(self.REQUIRED_PARAM_COURSE_ID, '')
             )
+            program_id = request.data.get(
+                self.REQUIRED_PARAM_PROGRAM_ID,
+                request.query_params.get(self.REQUIRED_PARAM_PROGRAM_ID, '')
+            )
             enterprise_customer_uuid = request.data.get(
                 self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER,
                 request.query_params.get(self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER)
             )
-        if not (username and course_id and enterprise_customer_uuid):
+        if not (username and (course_id or program_id) and enterprise_customer_uuid):
             raise ConsentAPIRequestError(
                 self.MISSING_REQUIRED_PARAMS_MSG.format(
                     username=username,
                     course_id=course_id,
                     enterprise_customer_uuid=enterprise_customer_uuid
                 )
             )
-        return username, course_id, enterprise_customer_uuid
+        return username, course_id, program_id, enterprise_customer_uuid
+
+    def get_no_record_response(self, request):
+        """
+        Get an HTTPResponse that can be used when there's no related EnterpriseCustomer.
+        """
+        username, course_id, program_id, enterprise_customer_uuid = self.get_required_query_params(request)
+        data = {
+            self.REQUIRED_PARAM_USERNAME: username,
+            self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER: enterprise_customer_uuid,
+            self.CONSENT_EXISTS: False,
+            self.CONSENT_GRANTED: False,
+            self.CONSENT_REQUIRED: False,
+        }
+        if course_id:
+            data[self.REQUIRED_PARAM_COURSE_ID] = course_id
+
+        if program_id:
+            data[self.REQUIRED_PARAM_PROGRAM_ID] = program_id
+
+        return Response(data, status=HTTP_200_OK)
```

Comment:
@douglashall requested that we switch to a 200 from the previous 404 we were throwing for nonexistent records.